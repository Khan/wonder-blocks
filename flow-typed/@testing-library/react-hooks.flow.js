// @flow
/**
 * This file is based on:
 * - https://github.com/testing-library/react-hooks-testing-library/blob/main/src/types/index.ts
 * - https://github.com/testing-library/react-hooks-testing-library/blob/main/src/types/react.ts
 * - https://github.com/testing-library/react-hooks-testing-library/blob/main/src/pure.ts
 */
 declare module "@testing-library/react-hooks" {
    declare export type Renderer<TProps> = {|
        render: (props?: TProps) => void,
        rerender: (props?: TProps) => void,
        unmount: () => void,
        act: Act,
    |};

    declare export type ServerRenderer<TProps> = {|
        ...Renderer<TProps>,
        hydrate: () => void,
    |};

    declare export type RendererProps<TProps, TResult> = {|
        callback: (props: TProps) => TResult,
        setError: (error: Error) => void,
        setValue: (value: TResult) => void,
    |};

    declare export type CreateRenderer<
        TProps,
        TResult,
        TRendererOptions: {...},
        TRenderer: Renderer<TProps>,
    > = (
        props: RendererProps<TProps, TResult>,
        options: TRendererOptions,
    ) => TRenderer;

    declare export type RenderResult<TValue> = {|
        +all: Array<TValue | Error>,
        +current: TValue,
        +error?: Error,
    |};

    declare export type ResultContainer<TValue> = {|
        result: RenderResult<TValue>,
    |};

    declare export type WaitOptions = {|
        interval?: number | false,
        timeout?: number | false,
    |};

    declare export type WaitForOptions = WaitOptions;
    declare export type WaitForValueToChangeOptions = WaitOptions;
    declare export type WaitForNextUpdateOptions = WaitOptions["timeout"];

    declare export type WaitFor = (
        callback: () => boolean | void,
        options?: WaitForOptions,
    ) => Promise<void>;
    declare export type WaitForValueToChange = (
        selector: () => mixed,
        options?: WaitForValueToChangeOptions,
    ) => Promise<void>;
    declare export type WaitForNextUpdate = (
        options?: WaitForNextUpdateOptions,
    ) => Promise<void>;

    declare export type AsyncUtils = {|
        waitFor: WaitFor,
        waitForValueToChange: WaitForValueToChange,
        waitForNextUpdate: WaitForNextUpdate,
    |};

    declare export type RenderHookResult<
        TProps,
        TValue,
        TRenderer: Renderer<TProps> = Renderer<TProps>,
    > = {|
        ...ResultContainer<TValue>,
        // Omit<Renderer<TProps>, 'render' | 'act'>
        ...$Diff<
            Renderer<TProps>,
            {|
                render: Renderer<TProps>["render"],
                act: Renderer<TProps>["act"],
            |},
        >,
        // Omit<TRenderer, keyof Renderer<TProps>>
        ...$Diff<TRenderer, Renderer<TProps>>,
        ...AsyncUtils,
    |};

    declare export type ServerRenderHookResult<
        TProps,
        TValue,
        TRenderer: ServerRenderer<TProps> = ServerRenderer<TProps>,
    > = RenderHookResult<TProps, TValue, TRenderer>;

    declare export type Act = {|
        (callback: () => Promise<void>): Promise<void>,
        (callback: () => void): void,
    |};

    declare export type CleanupCallback = () => Promise<void> | void;

    declare export type WrapperComponent<TProps> = React$ComponentType<TProps>;

    declare export type RendererOptions<TProps> = {|
        wrapper?: WrapperComponent<TProps>,
    |};

    declare export type RenderHookOptions<TProps> = {|
        initialProps?: TProps,
        wrapper?: WrapperComponent<any>,
    |};

    declare export type ReactHooksRenderer = {|
        renderHook: <TProps, TResult>(
            callback: (props: TProps) => TResult,
            options?: RenderHookOptions<TProps>,
        ) => RenderHookResult<TProps, TResult>,
        act: Act,
        cleanup: () => Promise<void>,
        addCleanup: (callback: CleanupCallback) => () => void,
        removeCleanup: (callback: CleanupCallback) => void,
        suppressErrorOutput: () => () => void,
    |};

    declare export type ReactHooksServerRenderer = {|
        // Omit<ReactHooksRenderer, 'renderHook'>
        ...$Diff<
            ReactHooksRenderer,
            {|
                renderHook: ReactHooksRenderer["renderHook"],
            |},
        >,
        renderHook: <TProps, TResult>(
            callback: (props: TProps) => TResult,
            options?: RenderHookOptions<TProps>,
        ) => ServerRenderHookResult<TProps, TResult>,
    |};

    declare export function renderHook<TProps, TResult>(
        callback: (props: TProps) => TResult,
        options?: RenderHookOptions<TProps>,
    ): RenderHookResult<TProps, TResult>;
    declare export var act: Act;
    declare export function cleanup(): Promise<void>;
    declare export function addCleanup(callback: CleanupCallback): () => void;
    declare export function removeCleanup(callback: CleanupCallback): void;
    declare export function suppressErrorOutput(): () => void;
}
