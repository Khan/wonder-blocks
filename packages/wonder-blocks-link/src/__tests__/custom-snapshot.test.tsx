import * as React from "react";

import {render} from "@testing-library/react";
import LinkCore from "../components/link-core";
import Link from "../components/link";

const defaultHandlers = {
    onClick: () => void 0,
    onMouseEnter: () => void 0,
    onMouseLeave: () => void 0,
    onMouseDown: () => void 0,
    onMouseUp: () => void 0,
    onTouchStart: () => void 0,
    onTouchEnd: () => void 0,
    onTouchCancel: () => void 0,
    onKeyDown: () => void 0,
    onKeyUp: () => void 0,
    onFocus: () => void 0,
    onBlur: () => void 0,
    tabIndex: 0,
} as const;

describe("Link", () => {
    test.each`
        tabIndex
        ${-1}
        ${0}
        ${1}
    `("<Link tabIndex={$tabIndex}>", ({tabIndex}: any) => {
        const {container} = render(
            <Link href="#" tabIndex={tabIndex}>
                Click me
            </Link>,
        );
        expect(container).toMatchSnapshot();
    });
});

describe("LinkCore", () => {
    for (const kind of ["primary", "secondary"] as const) {
        for (const href of ["#", "#non-existent-link"]) {
            for (const light of kind === "primary" ? [true, false] : [false]) {
                for (const visitable of kind === "primary"
                    ? [true, false]
                    : [false]) {
                    for (const inline of [true, false]) {
                        for (const state of ["focused", "hovered", "pressed"]) {
                            const stateProps = {
                                focused: state === "focused",
                                hovered: state === "hovered",
                                pressed: state === "pressed",
                                waiting: false,
                            } as const;
                            test(`kind:${kind} href:${href} light:${String(
                                light,
                            )} visitable:${String(visitable)} ${state}`, () => {
                                const {container} = render(
                                    <LinkCore
                                        href="#"
                                        inline={inline}
                                        kind={kind}
                                        light={light}
                                        visitable={visitable}
                                        {...stateProps}
                                        {...defaultHandlers}
                                    >
                                        Click me
                                    </LinkCore>,
                                );

                                expect(container).toMatchSnapshot();
                            });
                        }
                    }
                }
            }
        }
    }
});
