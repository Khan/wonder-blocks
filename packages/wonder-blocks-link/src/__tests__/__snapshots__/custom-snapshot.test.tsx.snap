// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link <Link tabIndex={-1}> 1`] = `
<div>
  <a
    class=""
    href="/#"
    style="cursor: pointer; text-decoration: none; outline: none; align-items: center; color: rgb(24, 101, 242);"
    tabindex="-1"
  >
    Click me
  </a>
</div>
`;

exports[`Link <Link tabIndex={0}> 1`] = `
<div>
  <a
    class=""
    href="/#"
    style="cursor: pointer; text-decoration: none; outline: none; align-items: center; color: rgb(24, 101, 242);"
    tabindex="0"
  >
    Click me
  </a>
</div>
`;

exports[`Link <Link tabIndex={1}> 1`] = `
<div>
  <a
    class=""
    href="/#"
    style="cursor: pointer; text-decoration: none; outline: none; align-items: center; color: rgb(24, 101, 242);"
    tabindex="1"
  >
    Click me
  </a>
</div>
`;

exports[`LinkCore kind:primary href:# light:false visitable:false focused 1`] = `
<div>
  <a
    class=""
    href="#"
    style="cursor: pointer; text-decoration: underline currentcolor solid; outline: none; align-items: center; color: rgb(24, 101, 242);"
    tabindex="0"
  >
    Click me
  </a>
</div>
`;

exports[`LinkCore kind:primary href:# light:false visitable:false focused 2`] = `
<div>
  <a
    class=""
    href="#"
    style="cursor: pointer; text-decoration: none; outline: none; align-items: center; color: rgb(24, 101, 242);"
    tabindex="0"
  >
    Click me
  </a>
</div>
`;

exports[`LinkCore kind:primary href:# light:false visitable:false hovered 1`] = `
<div>
  <a
    class=""
    href="#"
    style="cursor: pointer; text-decoration: underline currentcolor solid; outline: none; align-items: center; color: rgb(24, 101, 242);"
    tabindex="0"
  >
    Click me
  </a>
</div>
`;

exports[`LinkCore kind:primary href:# light:false visitable:false hovered 2`] = `
<div>
  <a
    class=""
    href="#"
    style="cursor: pointer; text-decoration: underline currentcolor solid; outline: none; align-items: center; color: rgb(24, 101, 242);"
    tabindex="0"
  >
    Click me
  </a>
</div>
`;

exports[`LinkCore kind:primary href:# light:false visitable:false pressed 1`] = `
<div>
  <a
    class=""
    href="#"
    style="cursor: pointer; text-decoration: underline currentcolor solid; outline: none; align-items: center; color: rgb(27, 80, 179);"
    tabindex="0"
  >
    Click me
  </a>
</div>
`;

exports[`LinkCore kind:primary href:# light:false visitable:false pressed 2`] = `
<div>
  <a
    class=""
    href="#"
    style="cursor: pointer; text-decoration: underline currentcolor solid; outline: none; align-items: center; color: rgb(27, 80, 179);"
    tabindex="0"
  >
    Click me
  </a>
</div>
`;

exports[`LinkCore kind:primary href:# light:false visitable:true focused 1`] = `
<div>
  <a
    class=""
    href="#"
    style="cursor: pointer; text-decoration: underline currentcolor solid; outline: none; align-items: center; color: rgb(24, 101, 242);"
    tabindex="0"
  >
    Click me
  </a>
</div>
`;

exports[`LinkCore kind:primary href:# light:false visitable:true focused 2`] = `
<div>
  <a
    class=""
    href="#"
    style="cursor: pointer; text-decoration: none; outline: none; align-items: center; color: rgb(24, 101, 242);"
    tabindex="0"
  >
    Click me
  </a>
</div>
`;

exports[`LinkCore kind:primary href:# light:false visitable:true hovered 1`] = `
<div>
  <a
    class=""
    href="#"
    style="cursor: pointer; text-decoration: underline currentcolor solid; outline: none; align-items: center; color: rgb(24, 101, 242);"
    tabindex="0"
  >
    Click me
  </a>
</div>
`;

exports[`LinkCore kind:primary href:# light:false visitable:true hovered 2`] = `
<div>
  <a
    class=""
    href="#"
    style="cursor: pointer; text-decoration: underline currentcolor solid; outline: none; align-items: center; color: rgb(24, 101, 242);"
    tabindex="0"
  >
    Click me
  </a>
</div>
`;

exports[`LinkCore kind:primary href:# light:false visitable:true pressed 1`] = `
<div>
  <a
    class=""
    href="#"
    style="cursor: pointer; text-decoration: underline currentcolor solid; outline: none; align-items: center; color: rgb(27, 80, 179);"
    tabindex="0"
  >
    Click me
  </a>
</div>
`;

exports[`LinkCore kind:primary href:# light:false visitable:true pressed 2`] = `
<div>
  <a
    class=""
    href="#"
    style="cursor: pointer; text-decoration: underline currentcolor solid; outline: none; align-items: center; color: rgb(27, 80, 179);"
    tabindex="0"
  >
    Click me
  </a>
</div>
`;

exports[`LinkCore kind:primary href:# light:true visitable:false focused 1`] = `
<div>
  <a
    class=""
    href="#"
    style="cursor: pointer; text-decoration: underline currentcolor solid; outline: none; align-items: center; color: rgb(255, 255, 255);"
    tabindex="0"
  >
    Click me
  </a>
</div>
`;

exports[`LinkCore kind:primary href:# light:true visitable:false focused 2`] = `
<div>
  <a
    class=""
    href="#"
    style="cursor: pointer; text-decoration: none; outline: none; align-items: center; color: rgb(255, 255, 255);"
    tabindex="0"
  >
    Click me
  </a>
</div>
`;

exports[`LinkCore kind:primary href:# light:true visitable:false hovered 1`] = `
<div>
  <a
    class=""
    href="#"
    style="cursor: pointer; text-decoration: underline currentcolor solid; outline: none; align-items: center; color: rgb(255, 255, 255);"
    tabindex="0"
  >
    Click me
  </a>
</div>
`;

exports[`LinkCore kind:primary href:# light:true visitable:false hovered 2`] = `
<div>
  <a
    class=""
    href="#"
    style="cursor: pointer; text-decoration: underline currentcolor solid; outline: none; align-items: center; color: rgb(255, 255, 255);"
    tabindex="0"
  >
    Click me
  </a>
</div>
`;

exports[`LinkCore kind:primary href:# light:true visitable:false pressed 1`] = `
<div>
  <a
    class=""
    href="#"
    style="cursor: pointer; text-decoration: underline currentcolor solid; outline: none; align-items: center; color: rgb(181, 206, 251);"
    tabindex="0"
  >
    Click me
  </a>
</div>
`;

exports[`LinkCore kind:primary href:# light:true visitable:false pressed 2`] = `
<div>
  <a
    class=""
    href="#"
    style="cursor: pointer; text-decoration: underline currentcolor solid; outline: none; align-items: center; color: rgb(181, 206, 251);"
    tabindex="0"
  >
    Click me
  </a>
</div>
`;

exports[`LinkCore kind:primary href:# light:true visitable:true focused 1`] = `
<div>
  <a
    class=""
    href="#"
    style="cursor: pointer; text-decoration: underline currentcolor solid; outline: none; align-items: center; color: rgb(255, 255, 255);"
    tabindex="0"
  >
    Click me
  </a>
</div>
`;

exports[`LinkCore kind:primary href:# light:true visitable:true focused 2`] = `
<div>
  <a
    class=""
    href="#"
    style="cursor: pointer; text-decoration: none; outline: none; align-items: center; color: rgb(255, 255, 255);"
    tabindex="0"
  >
    Click me
  </a>
</div>
`;

exports[`LinkCore kind:primary href:# light:true visitable:true hovered 1`] = `
<div>
  <a
    class=""
    href="#"
    style="cursor: pointer; text-decoration: underline currentcolor solid; outline: none; align-items: center; color: rgb(255, 255, 255);"
    tabindex="0"
  >
    Click me
  </a>
</div>
`;

exports[`LinkCore kind:primary href:# light:true visitable:true hovered 2`] = `
<div>
  <a
    class=""
    href="#"
    style="cursor: pointer; text-decoration: underline currentcolor solid; outline: none; align-items: center; color: rgb(255, 255, 255);"
    tabindex="0"
  >
    Click me
  </a>
</div>
`;

exports[`LinkCore kind:primary href:# light:true visitable:true pressed 1`] = `
<div>
  <a
    class=""
    href="#"
    style="cursor: pointer; text-decoration: underline currentcolor solid; outline: none; align-items: center; color: rgb(181, 206, 251);"
    tabindex="0"
  >
    Click me
  </a>
</div>
`;

exports[`LinkCore kind:primary href:# light:true visitable:true pressed 2`] = `
<div>
  <a
    class=""
    href="#"
    style="cursor: pointer; text-decoration: underline currentcolor solid; outline: none; align-items: center; color: rgb(181, 206, 251);"
    tabindex="0"
  >
    Click me
  </a>
</div>
`;

exports[`LinkCore kind:primary href:#non-existent-link light:false visitable:false focused 1`] = `
<div>
  <a
    class=""
    href="#"
    style="cursor: pointer; text-decoration: underline currentcolor solid; outline: none; align-items: center; color: rgb(24, 101, 242);"
    tabindex="0"
  >
    Click me
  </a>
</div>
`;

exports[`LinkCore kind:primary href:#non-existent-link light:false visitable:false focused 2`] = `
<div>
  <a
    class=""
    href="#"
    style="cursor: pointer; text-decoration: none; outline: none; align-items: center; color: rgb(24, 101, 242);"
    tabindex="0"
  >
    Click me
  </a>
</div>
`;

exports[`LinkCore kind:primary href:#non-existent-link light:false visitable:false hovered 1`] = `
<div>
  <a
    class=""
    href="#"
    style="cursor: pointer; text-decoration: underline currentcolor solid; outline: none; align-items: center; color: rgb(24, 101, 242);"
    tabindex="0"
  >
    Click me
  </a>
</div>
`;

exports[`LinkCore kind:primary href:#non-existent-link light:false visitable:false hovered 2`] = `
<div>
  <a
    class=""
    href="#"
    style="cursor: pointer; text-decoration: underline currentcolor solid; outline: none; align-items: center; color: rgb(24, 101, 242);"
    tabindex="0"
  >
    Click me
  </a>
</div>
`;

exports[`LinkCore kind:primary href:#non-existent-link light:false visitable:false pressed 1`] = `
<div>
  <a
    class=""
    href="#"
    style="cursor: pointer; text-decoration: underline currentcolor solid; outline: none; align-items: center; color: rgb(27, 80, 179);"
    tabindex="0"
  >
    Click me
  </a>
</div>
`;

exports[`LinkCore kind:primary href:#non-existent-link light:false visitable:false pressed 2`] = `
<div>
  <a
    class=""
    href="#"
    style="cursor: pointer; text-decoration: underline currentcolor solid; outline: none; align-items: center; color: rgb(27, 80, 179);"
    tabindex="0"
  >
    Click me
  </a>
</div>
`;

exports[`LinkCore kind:primary href:#non-existent-link light:false visitable:true focused 1`] = `
<div>
  <a
    class=""
    href="#"
    style="cursor: pointer; text-decoration: underline currentcolor solid; outline: none; align-items: center; color: rgb(24, 101, 242);"
    tabindex="0"
  >
    Click me
  </a>
</div>
`;

exports[`LinkCore kind:primary href:#non-existent-link light:false visitable:true focused 2`] = `
<div>
  <a
    class=""
    href="#"
    style="cursor: pointer; text-decoration: none; outline: none; align-items: center; color: rgb(24, 101, 242);"
    tabindex="0"
  >
    Click me
  </a>
</div>
`;

exports[`LinkCore kind:primary href:#non-existent-link light:false visitable:true hovered 1`] = `
<div>
  <a
    class=""
    href="#"
    style="cursor: pointer; text-decoration: underline currentcolor solid; outline: none; align-items: center; color: rgb(24, 101, 242);"
    tabindex="0"
  >
    Click me
  </a>
</div>
`;

exports[`LinkCore kind:primary href:#non-existent-link light:false visitable:true hovered 2`] = `
<div>
  <a
    class=""
    href="#"
    style="cursor: pointer; text-decoration: underline currentcolor solid; outline: none; align-items: center; color: rgb(24, 101, 242);"
    tabindex="0"
  >
    Click me
  </a>
</div>
`;

exports[`LinkCore kind:primary href:#non-existent-link light:false visitable:true pressed 1`] = `
<div>
  <a
    class=""
    href="#"
    style="cursor: pointer; text-decoration: underline currentcolor solid; outline: none; align-items: center; color: rgb(27, 80, 179);"
    tabindex="0"
  >
    Click me
  </a>
</div>
`;

exports[`LinkCore kind:primary href:#non-existent-link light:false visitable:true pressed 2`] = `
<div>
  <a
    class=""
    href="#"
    style="cursor: pointer; text-decoration: underline currentcolor solid; outline: none; align-items: center; color: rgb(27, 80, 179);"
    tabindex="0"
  >
    Click me
  </a>
</div>
`;

exports[`LinkCore kind:primary href:#non-existent-link light:true visitable:false focused 1`] = `
<div>
  <a
    class=""
    href="#"
    style="cursor: pointer; text-decoration: underline currentcolor solid; outline: none; align-items: center; color: rgb(255, 255, 255);"
    tabindex="0"
  >
    Click me
  </a>
</div>
`;

exports[`LinkCore kind:primary href:#non-existent-link light:true visitable:false focused 2`] = `
<div>
  <a
    class=""
    href="#"
    style="cursor: pointer; text-decoration: none; outline: none; align-items: center; color: rgb(255, 255, 255);"
    tabindex="0"
  >
    Click me
  </a>
</div>
`;

exports[`LinkCore kind:primary href:#non-existent-link light:true visitable:false hovered 1`] = `
<div>
  <a
    class=""
    href="#"
    style="cursor: pointer; text-decoration: underline currentcolor solid; outline: none; align-items: center; color: rgb(255, 255, 255);"
    tabindex="0"
  >
    Click me
  </a>
</div>
`;

exports[`LinkCore kind:primary href:#non-existent-link light:true visitable:false hovered 2`] = `
<div>
  <a
    class=""
    href="#"
    style="cursor: pointer; text-decoration: underline currentcolor solid; outline: none; align-items: center; color: rgb(255, 255, 255);"
    tabindex="0"
  >
    Click me
  </a>
</div>
`;

exports[`LinkCore kind:primary href:#non-existent-link light:true visitable:false pressed 1`] = `
<div>
  <a
    class=""
    href="#"
    style="cursor: pointer; text-decoration: underline currentcolor solid; outline: none; align-items: center; color: rgb(181, 206, 251);"
    tabindex="0"
  >
    Click me
  </a>
</div>
`;

exports[`LinkCore kind:primary href:#non-existent-link light:true visitable:false pressed 2`] = `
<div>
  <a
    class=""
    href="#"
    style="cursor: pointer; text-decoration: underline currentcolor solid; outline: none; align-items: center; color: rgb(181, 206, 251);"
    tabindex="0"
  >
    Click me
  </a>
</div>
`;

exports[`LinkCore kind:primary href:#non-existent-link light:true visitable:true focused 1`] = `
<div>
  <a
    class=""
    href="#"
    style="cursor: pointer; text-decoration: underline currentcolor solid; outline: none; align-items: center; color: rgb(255, 255, 255);"
    tabindex="0"
  >
    Click me
  </a>
</div>
`;

exports[`LinkCore kind:primary href:#non-existent-link light:true visitable:true focused 2`] = `
<div>
  <a
    class=""
    href="#"
    style="cursor: pointer; text-decoration: none; outline: none; align-items: center; color: rgb(255, 255, 255);"
    tabindex="0"
  >
    Click me
  </a>
</div>
`;

exports[`LinkCore kind:primary href:#non-existent-link light:true visitable:true hovered 1`] = `
<div>
  <a
    class=""
    href="#"
    style="cursor: pointer; text-decoration: underline currentcolor solid; outline: none; align-items: center; color: rgb(255, 255, 255);"
    tabindex="0"
  >
    Click me
  </a>
</div>
`;

exports[`LinkCore kind:primary href:#non-existent-link light:true visitable:true hovered 2`] = `
<div>
  <a
    class=""
    href="#"
    style="cursor: pointer; text-decoration: underline currentcolor solid; outline: none; align-items: center; color: rgb(255, 255, 255);"
    tabindex="0"
  >
    Click me
  </a>
</div>
`;

exports[`LinkCore kind:primary href:#non-existent-link light:true visitable:true pressed 1`] = `
<div>
  <a
    class=""
    href="#"
    style="cursor: pointer; text-decoration: underline currentcolor solid; outline: none; align-items: center; color: rgb(181, 206, 251);"
    tabindex="0"
  >
    Click me
  </a>
</div>
`;

exports[`LinkCore kind:primary href:#non-existent-link light:true visitable:true pressed 2`] = `
<div>
  <a
    class=""
    href="#"
    style="cursor: pointer; text-decoration: underline currentcolor solid; outline: none; align-items: center; color: rgb(181, 206, 251);"
    tabindex="0"
  >
    Click me
  </a>
</div>
`;

exports[`LinkCore kind:secondary href:# light:false visitable:false focused 1`] = `
<div>
  <a
    class=""
    href="#"
    style="cursor: pointer; text-decoration: underline currentcolor solid; outline: none; align-items: center; color: rgb(33, 36, 44);"
    tabindex="0"
  >
    Click me
  </a>
</div>
`;

exports[`LinkCore kind:secondary href:# light:false visitable:false focused 2`] = `
<div>
  <a
    class=""
    href="#"
    style="cursor: pointer; text-decoration: none; outline: none; align-items: center; color: rgba(33, 36, 44, 0.64);"
    tabindex="0"
  >
    Click me
  </a>
</div>
`;

exports[`LinkCore kind:secondary href:# light:false visitable:false hovered 1`] = `
<div>
  <a
    class=""
    href="#"
    style="cursor: pointer; text-decoration: underline currentcolor solid; outline: none; align-items: center; color: rgb(33, 36, 44);"
    tabindex="0"
  >
    Click me
  </a>
</div>
`;

exports[`LinkCore kind:secondary href:# light:false visitable:false hovered 2`] = `
<div>
  <a
    class=""
    href="#"
    style="cursor: pointer; text-decoration: underline currentcolor solid; outline: none; align-items: center; color: rgba(33, 36, 44, 0.64);"
    tabindex="0"
  >
    Click me
  </a>
</div>
`;

exports[`LinkCore kind:secondary href:# light:false visitable:false pressed 1`] = `
<div>
  <a
    class=""
    href="#"
    style="cursor: pointer; text-decoration: underline currentcolor solid; outline: none; align-items: center; color: rgb(27, 80, 179);"
    tabindex="0"
  >
    Click me
  </a>
</div>
`;

exports[`LinkCore kind:secondary href:# light:false visitable:false pressed 2`] = `
<div>
  <a
    class=""
    href="#"
    style="cursor: pointer; text-decoration: underline currentcolor solid; outline: none; align-items: center; color: rgb(33, 36, 44);"
    tabindex="0"
  >
    Click me
  </a>
</div>
`;

exports[`LinkCore kind:secondary href:#non-existent-link light:false visitable:false focused 1`] = `
<div>
  <a
    class=""
    href="#"
    style="cursor: pointer; text-decoration: underline currentcolor solid; outline: none; align-items: center; color: rgb(33, 36, 44);"
    tabindex="0"
  >
    Click me
  </a>
</div>
`;

exports[`LinkCore kind:secondary href:#non-existent-link light:false visitable:false focused 2`] = `
<div>
  <a
    class=""
    href="#"
    style="cursor: pointer; text-decoration: none; outline: none; align-items: center; color: rgba(33, 36, 44, 0.64);"
    tabindex="0"
  >
    Click me
  </a>
</div>
`;

exports[`LinkCore kind:secondary href:#non-existent-link light:false visitable:false hovered 1`] = `
<div>
  <a
    class=""
    href="#"
    style="cursor: pointer; text-decoration: underline currentcolor solid; outline: none; align-items: center; color: rgb(33, 36, 44);"
    tabindex="0"
  >
    Click me
  </a>
</div>
`;

exports[`LinkCore kind:secondary href:#non-existent-link light:false visitable:false hovered 2`] = `
<div>
  <a
    class=""
    href="#"
    style="cursor: pointer; text-decoration: underline currentcolor solid; outline: none; align-items: center; color: rgba(33, 36, 44, 0.64);"
    tabindex="0"
  >
    Click me
  </a>
</div>
`;

exports[`LinkCore kind:secondary href:#non-existent-link light:false visitable:false pressed 1`] = `
<div>
  <a
    class=""
    href="#"
    style="cursor: pointer; text-decoration: underline currentcolor solid; outline: none; align-items: center; color: rgb(27, 80, 179);"
    tabindex="0"
  >
    Click me
  </a>
</div>
`;

exports[`LinkCore kind:secondary href:#non-existent-link light:false visitable:false pressed 2`] = `
<div>
  <a
    class=""
    href="#"
    style="cursor: pointer; text-decoration: underline currentcolor solid; outline: none; align-items: center; color: rgb(33, 36, 44);"
    tabindex="0"
  >
    Click me
  </a>
</div>
`;
